client-config/src/main/java/com/palantir/conjure/java/client/config/ClientConfigurations.java

```diff
@@ -151,10 +149,10 @@ public static ProxySelector createProxySelector(ProxyConfiguration proxyConfig)
             case DIRECT:
                 return fixedProxySelectorFor(Proxy.NO_PROXY);
             case FROM_ENVIRONMENT:
-                String defaultEnvProxy = Preconditions.checkNotNull(
-                        System.getenv(ENV_HTTPS_PROXY),
-                        "Missing environment variable",
-                        SafeArg.of("name", ENV_HTTPS_PROXY));
+                String defaultEnvProxy = System.getenv(ENV_HTTPS_PROXY);
+                if (defaultEnvProxy == null) {
+                    return fixedProxySelectorFor(Proxy.NO_PROXY);
```

Comment:
Let's add an info level log line in either case so we can be confident we understand the environment when things go wrong.

Reply:
Done